tests/csapi/public_rooms_test.go

```diff
@@ -0,0 +1,188 @@
+package csapi_tests
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement"
+	"github.com/matrix-org/complement/client"
+	"github.com/matrix-org/complement/helpers"
+	"github.com/matrix-org/complement/match"
+	"github.com/matrix-org/complement/must"
+)
+
+func TestPublicRooms(t *testing.T) {
+	deployment := complement.Deploy(t, 1)
+	defer deployment.Destroy(t)
+
+	t.Run("parallel", func(t *testing.T) {
+		// sytest: Name/topic keys are correct
+		t.Run("Name/topic keys are correct", func(t *testing.T) {
+			t.Parallel()
+			authedClient := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+
+			// Define room configurations matching the Sytest
+			roomConfigs := []struct {
+				alias string
+				name  string
+				topic string
+			}{
+				{"publicroomalias_no_name", "", ""},
+				{"publicroomalias_with_name", "name_1", ""},
+				{"publicroomalias_with_topic", "", "topic_1"},
+				{"publicroomalias_with_name_topic", "name_2", "topic_2"},
+				{"publicroom_with_unicode_chars_name", "un nom français", ""},
+				{"publicroom_with_unicode_chars_topic", "", "un topic à la française"},
+				{"publicroom_with_unicode_chars_name_topic", "un nom français", "un topic à la française"},
+			}
+
+			// Create all rooms with their configurations
+			createdRooms := make(map[string]struct {
+				roomID string
+				name   string
+				topic  string
+			})
+
+			for _, config := range roomConfigs {
+				roomOptions := map[string]interface{}{
+					"visibility":      "public",
+					"room_alias_name": config.alias,
+				}
+
+				if config.name != "" {
+					roomOptions["name"] = config.name
+				}
+				if config.topic != "" {
+					roomOptions["topic"] = config.topic
+				}
+
+				roomID := authedClient.MustCreateRoom(t, roomOptions)
+				createdRooms[config.alias] = struct {
+					roomID string
+					name   string
+					topic  string
+				}{
+					roomID: roomID,
+					name:   config.name,
+					topic:  config.topic,
+				}
+				t.Logf("Created room %s with alias %s", roomID, config.alias)
+			}
+
+			// Poll /publicRooms until all our rooms appear with correct data
+			authedClient.MustDo(t, "GET", []string{"_matrix", "client", "v3", "publicRooms"},
+				client.WithRetryUntil(15*time.Second, func(res *http.Response) bool {
```

Comment:
Is 15 seconds necessary in real-world testing?

We could re-use `client.SyncUntilTimeout` which is usually 5 seconds for `register`/`login`.

For reference, I had a [10 second timeout](https://github.com/matrix-org/matrix-viewer/blob/5c448f24c252ebc4068824f1e1e47cefe7527b5f/test/test-utils/client-utils.js#L426-L431) in the Matrix Viewer tests but that might have been arbitrary as well.

Reply:
Sure. Looking at the latest run, the whole test took 850ms: https://github.com/matrix-org/complement/actions/runs/18659975767/job/53197931572#step:8:3607

So I think 5 or 10 seconds is appropriate. I'll re-use `SyncUnitTimeout` to it can be tweaked along with other timeouts..